src/query/sql/src/planner/semantic/type_check.rs

```diff
@@ -2955,3 +3010,42 @@ pub fn validate_function_arg(
         }
     }
 }
+
+// Some check functions for like expression
+fn check_const(like_str: &str) -> bool {
+    for char in like_str.chars() {
+        if char == '_' || char == '%' {
+            return false;
+        }
+    }
+    true
+}
+
+fn check_prefix(like_str: &str) -> bool {
+    if like_str.contains("\\%") {
+        return false;
+    }
+    if like_str.len() == 1 && matches!(like_str, "%" | "_") {
+        return false;
+    }
+    if like_str.chars().filter(|c| *c == '%').count() != 1 {
+        return false;
+    }
+
+    let mut i: usize = like_str.len();
+    while i > 0 {
+        if like_str.chars().nth(i - 1).unwrap() != '%' {
+            break;
+        }
+        i -= 1;
+    }
+    if i == like_str.len() {
+        return false;
+    }
+    for j in (0..i).rev() {
+        if like_str.chars().nth(j).unwrap() == '%' || like_str.chars().nth(j).unwrap() == '_' {
```

Comment:
Only need `like_str.chars().nth(j).unwrap() == '_'`, because the interval [0, i) has no `%`.